pkg/controller/datavolume-controller_test.go

```diff
@@ -1581,6 +1581,179 @@ var _ = Describe("All DataVolume Tests", func() {
 			Entry("should default to snapshot", nil, nil, cdiv1.CDICloneStrategy(cdiv1.CloneStrategySnapshot)),
 		)
 	})
+
+	var _ = Describe("Clone with empty storage size", func() {
+		scName := "testsc"
+		accessMode := []corev1.PersistentVolumeAccessMode{corev1.ReadOnlyMany}
+		sc := createStorageClassWithProvisioner(scName, map[string]string{
+			AnnDefaultStorageClass: "true",
+		}, "csi-plugin")
+
+		// detectCloneSize tests
+
+		It("Size-detection fails when source PVC is not attainable", func() {
+			dv := newCloneDataVolumeWithEmptyStorage("test-dv", "default")
+			cloneStrategy := cdiv1.CDICloneStrategy(cdiv1.CloneStrategyHostAssisted)
+			storageProfile := createStorageProfileWithCloneStrategy(scName, []cdiv1.ClaimPropertySet{
+				{AccessModes: accessMode, VolumeMode: &blockMode}}, &cloneStrategy)
+
+			reconciler := createDatavolumeReconciler(dv, storageProfile, sc)
+			pvcSpec, err := RenderPvcSpec(reconciler.client, reconciler.recorder, reconciler.log, dv)
+			Expect(err).ToNot(HaveOccurred())
+			err = reconciler.detectCloneSize(dv, pvcSpec, HostAssistedClone)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("Size-detection fails when source PVC is not fully imported", func() {
+			dv := newCloneDataVolumeWithEmptyStorage("test-dv", "default")
+			cloneStrategy := cdiv1.CDICloneStrategy(cdiv1.CloneStrategyHostAssisted)
+			storageProfile := createStorageProfileWithCloneStrategy(scName, []cdiv1.ClaimPropertySet{
+				{AccessModes: accessMode, VolumeMode: &blockMode}}, &cloneStrategy)
+
+			pvc := createPvcInStorageClass("test", metav1.NamespaceDefault, &scName, nil, nil, corev1.ClaimBound)
+			reconciler := createDatavolumeReconciler(dv, pvc, storageProfile, sc)
+
+			pvcSpec, err := RenderPvcSpec(reconciler.client, reconciler.recorder, reconciler.log, dv)
+			Expect(err).ToNot(HaveOccurred())
+			err = reconciler.detectCloneSize(dv, pvcSpec, HostAssistedClone)
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(Equal(ErrImportPVCNotReady))
```

Comment:
I don't think it should fail here, it should just return a size of 0. You could potentially record an event and check that in the unit test to ensure the right `0` is returned.